packages/jest-circus/src/__tests__/hooksError.test.ts

```diff
@@ -5,30 +5,33 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import {Circus} from '@jest/types';
+import type {Circus} from '@jest/types';
 import circus from '../';
 
-describe.each(['beforeEach', 'beforeAll', 'afterEach', 'afterAll'])(
-  '%s hooks error throwing',
-  (fn: Circus.HookType) => {
-    test.each([
-      ['String'],
-      [1],
-      [[]],
-      [{}],
-      [Symbol('hello')],
-      [true],
-      [null],
-      [undefined],
-    ])(
-      `${fn} throws an error when %p is provided as a first argument to it`,
-      el => {
-        expect(() => {
-          circus[fn](el);
-        }).toThrowError(
-          'Invalid first argument. It must be a callback function.',
-        );
-      },
-    );
-  },
-);
+describe.each([
+  'beforeEach',
+  'beforeAll',
+  'afterEach',
+  'afterAll',
+] as Array<Circus.HookType>)('%s hooks error throwing', fn => {
```

Comment:
No magic with `as const`?

```suggestion
] as const)('%s hooks error throwing', fn => {
```

Reply:
`as const` doesn't really work - then it's always hard coded to the first value

Reply:
probably an issue with `jest-each`'s types, but I haven't dug into it

Reply:
Just checked. Interesting... I will check what is going on.

Reply:
Yep. Types are buggy and test is broken too.

https://github.com/facebook/jest/blob/22322c9acc3197e43337b5415755823f09f410fe/packages/jest-types/__typetests__/each.test.ts#L31

Here `list` is `[1, 2, 3]`. That’s why it looks like just first value is picked up. Add second arg and second one will be there. In reality there is no second arg ;D Ups.. I will fix. Good to catch it!